### PR TITLE
[Fixed] AlienVault USM Anywhere fetch incident bug

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/17994

## Description
There were 2 bugs:

1. Wrong date format - Added a function that handles both supported date formats, that's to maintain backwards compatibility.
2. Stopped fetching incidents after two fetches of 100 incidents each - Added fetch_limit parameter to integration's configuration. There wasn't a problem with the logic of the fetch_incidents function, the problem was that the inventory of alarms wasn't sufficient and after fetching a certain amount of alarms, the inventory was over. I've decreased the fetch_limit from 100 to 10 and it works fine. @mgioia-demisto should configure AlienVault to support unlimited (or big amount of) alarms.

## Required version of Demisto
The change in code was tested with a Demisto server installed on an AWS machine that @mgioia-demisto provided. The version of Demisto is 4.5.0.
The instance that was configured and tested is AlienVault USM Anywhere_copy.

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Code Review

### CODE REVIEW

**Comment by guyfreund** [comment]
last commit msg [8891cd4](https://github.com/demisto/content/pull/4320/commits/8891cd457bd4a06dda52bfc58ad415ffaf4ca8bd) should be "added CHANGELOG.md file"

File: Integrations/AlienVault_USM_Anywhere/AlienVault_USM_Anywhere.py [hunk 1]
```diff
@@ -199,6 +204,34 @@ def parse_events(events_data):
...
+    :return: The integer representation of the key's value in the dict params
+    """
+    try:
+        if target_dict:
+            value = target_dict.get(key)
+            if value:
+                target_dict[key] = int(value)
+                return target_dict[key]
+    except ValueError:
+        raise Exception(f'This value for {key} must be an integer.')
```
Comment (yaakovi) [inline Integrations/AlienVault_USM_Anywhere/AlienVault_USM_Anywhere.py]:
```suggestion
        raise ValueError(f'The value for {key} must be an integer.')
```

File: Integrations/AlienVault_USM_Anywhere/AlienVault_USM_Anywhere.py [hunk 2]
```diff
@@ -199,6 +204,34 @@ def parse_events(events_data):
...
+
+
+def run_function_date_format(function, arg):
+    """
+    :param function: A function that has the arg date_format
+    :param arg: An argument necessary for the function
+    :return: The result of the application expression
+    """
+    try:
```
Comment (yaakovi) [inline Integrations/AlienVault_USM_Anywhere/AlienVault_USM_Anywhere.py]: lets convert this to a loop in order to support more future formats without a lot of changes

**Comment by guyfreund** [comment]
Hi @kirbles19, please review the added param Time Format. Lines changed in yml file 47-48.
All other changes changed either in backend / already been approved in ZeroFox Integration.

File: Integrations/AlienVault_USM_Anywhere/AlienVault_USM_Anywhere.py [hunk 3]
```diff
@@ -199,6 +219,22 @@ def parse_events(events_data):
...
+    :return: The integer representation of the key's value in the dict params
+    """
+    try:
+        if target_dict:
+            value = target_dict.get(key)
+            if value:
+                target_dict[key] = int(value)
+                return target_dict[key]
+    except ValueError:
+        raise ValueError(f'This value for {key} must be an integer.')
```
Comment (yaakovi) [inline Integrations/AlienVault_USM_Anywhere/AlienVault_USM_Anywhere.py]: you did not fix this one...
```suggestion
        raise ValueError(f'The value for {key} must be an integer.')
```

**Comment by kirbles19** [comment]
@guyfreund Done reviewing .yml and .md.

**Comment by yaakovi** [comment]
@guyfreund fix the conflicts and also add this to the `CHANGELOG.md`:

```
- Added the *Fetch limit* and *Time format* parameters to the instance configuration.
```

@kirbles19 FYI

**Comment by yardensade** [comment]
Delete merged branches after you are done with them.

**Pull request merged**